quorum/src/quorum.rs

```
@@ -0,0 +1,176 @@
+         Ok(eligible_claims) => eligible_claims,
+         Err(e) => return Err(e),
+      };
+      
+      let elected_quorum = match self.get_final_quorum(quorum_seed, eligible_claims, nodes){
+         Ok(elected_quorum) => elected_quorum,
+         Err(e) => return Err(e),
+      };
+      return Ok(elected_quorum);
+  }
+
+
+  fn nonce_up_claims(claims: Vec<Claim>) -> Vec<Claim>{
+      let mut nonce_up_claims = Vec::new();
+      claims.iter().for_each(|claim|{
+         let mut nonce_up_claim = claim.clone();
+         nonce_up_claim.nonce += 1;
+         nonce_up_claims.push(nonce_up_claim);
+      });
+      return nonce_up_claims;
+  }
+
+  fn run_election(&mut self, child_block: &DummyChildBlock, claims: Vec<Claim>, nodes: Vec<DummyNode>) -> Result<&Quorum, InvalidQuorum>{
+     match self.elect_quorum(child_block, claims, nodes){
+        Ok(quorum) => return Ok(quorum),
+        Err(e) => return Err(e),
+     };
+  }
+}
+
+//result enum for errors
+impl Quorum{
+  //make new generate a blank/default quorum like a constructor
+  pub fn new() -> Quorum{
+     return Quorum{
+        quorum_seed: 0,
+        masternodes: Vec::new(),
+        quorum_pk: String::new(),
+        election_block_height: 0,
+        election_timestamp: 0,
+     } 
+  }
+
+  pub fn generate_quorum_seed(&mut self, child_block: &DummyChildBlock) -> Result<u128, InvalidQuorum>{
+
+     let child_block_timestamp: u128 = child_block.timestamp;
+     let child_block_height: u128 = child_block.height;
+
+     if child_block_height == 0{
+        dbg!("test 2 fails here");
+        return Err(InvalidQuorum::InvalidChildBlockError());
+     } else if child_block_timestamp == 0 {
+        dbg!("test 3 fails here");
+        return Err(InvalidQuorum::InvalidChildBlockError());
+     } else {
+        let sk = VVRF::generate_secret_key();
+        let mut vvrf = VVRF::new(child_block.hash.as_bytes(), sk);
+     
+        assert!(VVRF::verify_seed(&mut vvrf).is_ok());
```

Let's replace this with an early return of an `Err` if this fails, if the `assert!` fails it'll cause the thread to `panic` which we don't want, we just want to propagate the error so it can be handled.